x-pack/plugins/security_solution/server/usage/queries/get_event_log_by_type_and_status.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger, SavedObjectsFindResult } from 'kibana/server';
+import {
+  EQL_RULE_TYPE_ID,
+  INDICATOR_RULE_TYPE_ID,
+  ML_RULE_TYPE_ID,
+  QUERY_RULE_TYPE_ID,
+  THRESHOLD_RULE_TYPE_ID,
+  SAVED_QUERY_RULE_TYPE_ID,
+} from '@kbn/securitysolution-rules';
+import type { EventLogTypeStatusAggs, RuleSearchResult } from '../types';
+import type { EventLogStatusMetric } from '../detections/rules/types';
+import { getEventLogAggByStatuses } from './utils/get_event_log_agg_by_statuses';
+import { transformEventLogTypeStatus } from './utils/transform_event_log_type_status';
+import { getInitialEventLogUsage } from '../detections/rules/get_initial_usage';
+import { getSearchForAllRules } from './utils/get_search_for_all_rules';
+import { getSearchForElasticRules } from './utils/get_search_for_elastic_rules';
+import { getSearchForCustomRules } from './utils/get_search_for_custom_rules';
+
+export interface GetEventLogByTypeAndStatusOptions {
+  esClient: ElasticsearchClient;
+  eventLogIndex: string;
+  logger: Logger;
+  ruleResults: Array<SavedObjectsFindResult<RuleSearchResult>>;
+}
+
+/**
+ * Gets the event logs by their rule type and rule status. Returns the structure
+ * transformed. If it malfunctions or times out then it will not malfunction other
+ * parts of telemetry.
+ * NOTE: This takes in "ruleResults" to filter against elastic rules and custom rules.
+ * If the event log recorded information about which rules were elastic vs. custom this
```

Comment:
Yeah, would be nice if we logged more metadata about the rule itself so there's not so much overhead in processing like this.